projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,476 @@
+
+This component displays a list of *Nano Contract Transaction* items. Each item represents a transaction as either an input (sent), an output (received), or a method execution (called). The list is designed to provide essential transaction information in a clear and concise manner.
+
+Tapping on an item opens the *Nano Contract Transaction Details* component, providing further information on the selected transaction item.
+
+##### Design Bill
+- **Placement**:
+	- The *Nano Contract Transaction List* should be displayed within the body section of the *Nano Contract Transactions* screen.
+
+- **Functionality**:
+	- Each item in the list is actionable, leading to detailed transaction information upon selection.
+	- Implement an infinite scrolling view to display a potentially large number of transactions.
+	- Each transaction item must contain *Description*, *Token Amount*, *Token Symbol*, *Blueprint Method Name* and *Date and Time* information for the item list view.
+
+- **Visibility Conditions**:
+	- Differentiate item types *sent*, *received*, and *called* using distinct visual indicators.
+	- A *called* item type must not contain *Token Amount* and *Token Symbol* information. 
+
+>[!NOTE]
+>Example of item description per type:
+>- "You *received* `<Token Symbol>` from `<Blueprint Method Name>`"
+>- "You *sent* `<Token Symbol>` to `<Blueprint Method Name>`"
+>- "You *called* `<Blueprint Method Name>`"
+
+### Component: Nano Contract Transaction Details
+
+This component provides detailed information for each transaction item type: *sent*, *received*, and *called*.
+
+For *sent* and *received* transaction items, the header displays *Token Amount* and *Token Symbol*, with an optional label *Amount* underneath them.
+
+The body section includes:
+- *Token ID*
+- *Description*
+- *Blueprint Method Name*
+- *Date and Time*
+- *Transaction ID*
+- *Public Explorer*
+
+Taping on *Public Explorer* manigates the user to the transaction page in the explorer website.
+
+##### Design Bill
+- **Placement**:
+	- Position the header at the top of the component with clear visual separation from the body.
+	- The body should follow immediately after the header, occupying the main portion of the component.
+
+- **Functionality**:
+	- The *Public Explorer* action must be visually distinct and recognizable as an actionable item.
+	- It should be easy for the user to dismiss the details component and focus back in the list view.
+
+- **Visibility Conditions**:
+	- Display header information of *Token Amount* and *Token Symbol* only for *sent* and *received* transaction items.
+	- For *called* transaction items, only the body section should be visible.
+	- Differentiate item *Public Explorer* using distinct visual indicators.
+  
+## Registering Nano Contract
+
+```mermaid
+graph LR
+    A[Main] -->B[Nano Contract Transactions]
+    A -->|This Flow| C[Nano Contract Registration]
+```
+
+### Screen: Nano Contract Registration
```

Comment:
I missed some context on what is a nano contract registration, can you add a short description on what registering a nano contract means?

Reply:
Sure! I'll add it.

By registering a Nano Contract the wallet can load its transactions.